agent/stats/queue.go

```diff
@@ -215,3 +249,51 @@ func (queue *Queue) getCWStatsSet(f getUsageFunc) (*ecstcs.CWStatsSet, error) {
 		Sum:         &sum,
 	}, nil
 }
+
+// getULongStatsSet gets the stats set for the specified raw stat type
+// stats come from docker as uint64 type, and by neccesity are packed into int64 type
+// where there is overflow (math.MaxInt64 + 1 or greater)
+// we capture the excess in optional overflow fields.
+func (queue *Queue) getULongStatsSet(f getUsageIntFunc) (*ecstcs.ULongStatsSet, error) {
+	queue.lock.Lock()
+	defer queue.lock.Unlock()
+
+	queueLength := len(queue.buffer)
+	if queueLength < 2 {
+		// Need at least 2 data points to calculate this.
+		return nil, fmt.Errorf("No data in the queue")
+	}
+
+	var min, max, sum uint64
+	var sampleCount int64
+	min = math.MaxUint64
+	max = 0
+	sum = 0
+	sampleCount = 0
+
+	for _, stat := range queue.buffer {
+		thisStat := f(&stat)
+		if thisStat < min {
+			min = thisStat
+		}
+		if thisStat > max {
+			max = thisStat
+		}
+		sum += thisStat
+		sampleCount++
```

Comment:
Do we need to calculate this sampleCount? Could we use queueLength?

Reply:
as I see it, either is fine.  I wanted to duplicate what we do in cwstats function above.